app/models/bulkrax/csv_entry.rb

```diff
@@ -22,7 +22,9 @@ def self.read_data(path)
     def self.data_for_entry(data)
       # If a multi-line CSV data is passed, grab the first row
       data = data.first if data.is_a?(CSV::Table)
+      raw_model = data[:model]
       raw_data = data.to_h
+      raw_data[:model] = raw_model
```

Comment:
`raw_data_model_data[:model] = raw_data[:model].data` ;-) 

looks like `raw_model` never gets used anywhere else and this code feels confusing to me. `data` still exists unchanged after the to_h, so maybe 

```
raw_data[:model] = data[:model]
```

and a comment as to why?

Reply:
This was a really weird case because it was a CSV element. when it uses to_h it sometimes doesn't translate the model correctly. So I pulled it out first then after it does to_h I overwrite the possibly incorrect model with the correct one. I'll add a comment explaining this.